Sources/NIO/Bootstrap.swift

```diff
@@ -967,3 +999,146 @@ public final class NIOPipeBootstrap {
         }
     }
 }
+
+/// An channel option which can be applied to bootstrap using shorthand notation.
+/// - See: ServerBootstrap.serverOptions(_ options: [ShorthandBootstrapOption])
+public struct ShorthandServerBootstrapOption {
+    private let data: ShorthandServerOption
+    
+    private init(_ data: ShorthandServerOption) {
+        self.data = data
+    }
+    
+    /// Apply the contained option to the supplied ServerBootstrap
+    /// - Parameter serverBootstrap: bootstrap to apply this option to.
+    /// - Returns: the modified bootstrap (currently the same one mutated)
+    func applyOption(to serverBootstrap : ServerBootstrap) {
+        data.applyOption(to: serverBootstrap)
+    }
+    
+    private enum ShorthandServerOption {
+        case ReuseAddr(Bool)
+        case AutoRead(Bool)
+        case Backlog(Int32)
+        
+        func applyOption(to serverBootstrap: ServerBootstrap) {
+            switch self {
+                case .AutoRead(let value):
+                    let _ = serverBootstrap.serverChannelOption(ChannelOptions.autoRead, value: value)
+                case .ReuseAddr(let value):
+                    let _ = serverBootstrap.serverChannelOption(ChannelOptions.socketOption(.reuseaddr), value: value ? 1 : 0)
+                case .Backlog(let value):
+                    let _ = serverBootstrap.serverChannelOption(ChannelOptions.backlog, value: value)
+            }
+        }
+    }
+}
+
+/// Approved shorthand server options.
+extension ShorthandServerBootstrapOption {
+    /// Option to reuse address.
+    /// - See:  NIOBSDSocket.Option.reuseaddr
+    public static let reuseAddr = ShorthandServerBootstrapOption(.ReuseAddr(true))
+    
+    /// Option to disble autoRead
+    /// - See: ChannelOptions.autoRead
+    public static let disableAutoRead = ShorthandServerBootstrapOption(.AutoRead(false))
+    
+    /// `BacklogOption` allows users to configure the `backlog` value as specified in `man 2 listen`. This is only useful for `ServerSocketChannel`s.
+    /// - See: ChannelOptions.backlog
+    public static func backlog(_ value : ChannelOptions.Types.BacklogOption.Value) -> ShorthandServerBootstrapOption {
+        return ShorthandServerBootstrapOption(.Backlog(value))
+    }
+}
+
+/// An channel option which can be applied to bootstrap using shorthand notation.
+/// - See: ServerBootstrap.serverOptions(_ options: [ShorthandBootstrapOption])
+public struct ShorthandChildBootstrapOption {
+    private let data: ShorthandChildOption
+    
+    private init(_ data: ShorthandChildOption) {
+        self.data = data
+    }
+    
+    /// Apply the contained option to the supplied ServerBootstrap
+    /// - Parameter serverBootstrap: bootstrap to apply this option to.
+    /// - Returns: the modified bootstrap (currently the same one mutated)
+    func applyOption(to serverBootstrap : ServerBootstrap) {
+        data.applyOption(to: serverBootstrap)
+    }
+    
+    private enum ShorthandChildOption {
```

Comment:
If we wanted to we could try to re-use this backing enum, and just only use cases in each public structure that apply to the relevant bootstrap. Up to you: I'm not sure if the reduced repetition outweighs the risks for setting the wrong option.

Reply:
Decided not to reuse.